grails-app/services/au/org/ala/ecodata/MetadataService.groovy

```diff
@@ -1275,6 +1277,52 @@ class MetadataService {
 
     }
 
+    List<Term> findTermsByCategory(String category, String hubId = null) {
+        List<Term> terms = Term.createCriteria().list {
+            if (hubId) { // Support categories that are used in multiple hubs
+                eq('hubId', hubId)
+            }
+            eq('category', category)
+            ne('status', Status.DELETED)
+            order('term', 'asc')
+        }
+        terms
+    }
+
+    Term deleteTerm(String termId) {
+
+        Term term = Term.findByTermIdAndStatus(termId, Status.ACTIVE)
+        if (!term) {
+            return null
+        }
+
+        // Because of the unique constraint on Terms, if the same Term is deleted then
+        // re-added we can't soft delete it a second time without hard deleting any existing
+        // soft deleted Term.
+        Term deletedTerm = Term.findByTermAndCategoryAndHubIdAndStatus(
+                term.term, term.category, term.hubId, Status.DELETED)
+        if (deletedTerm) {
+            deletedTerm.delete(flush:true)
+        }
 
+        term.status = Status.DELETED
+        term.save(flush:true)
+
+        term
+    }
+
+    Term updateTerm(Map termProps) {
+        Term term
+        if (!termProps?.termId) {
+            term = new Term()
+        }
+        else {
+            term = Term.findByTermId(termProps.termId)
+        }
+
+        bindData(term, termProps)
```

Comment:
Just wondering if termId needs to be added to exclude list?

Reply:
I can add it but I'm not sure it will have any effect as we are using the termId id from the props to find the Term to update anyway (so binding it will necessarily be using the same termId anyway).
Or do you think we should be putting the termId in the URL path and excluding it from the properties binding?

Reply:
Yes, sounds good. No need.